segmentation/utils/segmentation_utils.py

```diff
@@ -278,6 +278,86 @@ def combine_segmentation_masks(big_mask_xr, small_mask_xr, size_threshold,
         format="NETCDF3_64BIT")
 
 
+def find_nuclear_mask_id(nuc_segmentation_mask, cell_coords):
+    """Get the ID of the nuclear mask which has the greatest amount of overlap with a given cell
+
+    Args:
+        nuc_segmentation_mask: label mask of nuclear segmentations
+        cell_coords: list of coords specifying pixels that belong to a cell
+
+    Returns:
+        nuclear_mask_id: ID of the nuclear mask that overlaps most with cell. If not matches found,
+            returns None.
+    """
+
+    ids, counts = np.unique(nuc_segmentation_mask[tuple(cell_coords.T)], return_counts=True)
+
+    max_id = ids[np.argmax(counts)]
+
+    if max_id == 0:
+        if len(ids) == 1:
+            nuclear_mask_id = None
+        else:
+            non_zero_idx = ids > 0
+            ids, counts = ids[non_zero_idx], counts[non_zero_idx]
+
+            nuclear_mask_id = ids[np.argmax(counts)]
+    else:
+        nuclear_mask_id = max_id
```

Comment:
I think this logic can be cleaned up with something like:
```
# np.unique returns in sorted order, so this works assuming all positive ids
if ids[1:]:
  nuclear_mask_id = ids[np.argmax(counts[1:]) + 1]
else:
  nuclear_mask_id = None
```
I think this could even be a one-line ternary statement, but that might look a little messy

Reply:
This doesn't work for me: 
![image](https://user-images.githubusercontent.com/13770365/87099029-9b067600-c1fd-11ea-8050-0103582d7a00.png)

Reply:
oh that should be something like `if x[1:] is not None`, sorry; all it means to do is check to see if the non-zero part of the array is zero

Reply:
Oooh, got it. That's much neater. However, I don't think it works in the case that there's only a single ID and it's not zero, right?

Reply:
possibly not; idk if it's bad practice, but you could maybe wrap it in some sort of try/catch instead?  Some thing like:
```
try:
  nuclear_mask_id = ids[np.argmax(counts[1:])+1]
except:
  nuclear_mask_id = None
```

Reply:
practice-wise, it looks like this is fine as long as except isn't blanket-ed.  So using `except ValueError as e:` is best.  Pretty sure it'll always be a `ValueError` from calling `np.argmax` on an empty list; there's a possibility it'll be an `IndexError` if the size of `counts` is somehow not the same as `ids`, but I don't think we want that behavior anyways

Reply:
Okay, in that case I'm going to leave it as is. We want to keep the functionality of a single non-zero value being accurately reported as the nuclear_id, and not raise an exception.

Reply:
oooh, nvm.  I misunderstood what you were saying, sorry.  I think the fix is:
```
if ids[ids != 0] is not None:
  nuclear_mask_id = ids[np.argmax(counts[ids != 0])]
else:
  nuclear_mask_id = None
```
this could also be replaced with the try/except structure as well.  But this is probably clearer syntax.